src/libraries/System.Text.RegularExpressions/gen/RegexGenerator.Emitter.cs

```diff
@@ -837,7 +837,7 @@ void EmitFixedSet_LeftToRight()
                                 writer.WriteLine($"// The primary set being searched for was found. {numRemainingSets} more set{(numRemainingSets > 1 ? "s" : "")} will be checked so as");
                                 writer.WriteLine($"// to minimize the number of places TryMatchAtCurrentPosition is run unnecessarily.");
                                 writer.WriteLine($"// Make sure {(numRemainingSets > 1 ? "they fit" : "it fits")} in the remainder of the input.");
-                                using (EmitBlock(writer, $"if ((uint)(i + {maxDistance}) >= span.Length)"))
+                                using (EmitBlock(writer, $"if ((uint)(i + {maxDistance}) >= (uint)span.Length)"))
```

Comment:
Shouldn't we also change the compiler code to add a `conv.i8` IL instruction to perform the cast here:

[https://github.com/dotnet/runtime/blob/844b29856a32dd178214cc57dc2df60974c10fa4/src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompiler.cs#L921](https://github.com/dotnet/runtime/blob/844b29856a32dd178214cc57dc2df60974c10fa4/src/libraries/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexCompiler.cs#L921)

It also looks like we are not currently performing the cast in `i + maxdistance` in there today so we probably should.

Reply:
Why would we need conv.i8?

The comparison is being performed as unsigned:
> _ilg!.Emit(OpCodes.Bge_Un, returnFalse);

Reply:
Oh, I missed that, never mind then.